src/generic/DirectSum.jl

```diff
@@ -216,3 +223,31 @@ end
 function DirectSum(vals::AbstractAlgebra.FPModule{T}...) where T <: RingElement
    return DirectSum([vals...])
 end
+
+function hom(D::DirectSumModule{T}, A::AbstractAlgebra.FPModule{T}, m::Vector{<:ModuleHomomorphism{T}}) where T <: RingElement
```

Comment:
What is hom for? We already had ModuleHomomorphism, so I assume this is something else. Docs required of course.

Reply:
And tests for the nontrivial ones.